libs/external-data/src/readability/readability.service.ts

```diff
@@ -0,0 +1,220 @@
+import { DbService } from '@dua-upd/db';
+import { Injectable } from '@nestjs/common';
+import { wait } from '@dua-upd/utils-common';
+import { lexiqueLookup } from './utils';
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import axios from 'axios';
+import { load } from 'cheerio/lib/slim';
+import readability from 'text-readability';
+import stopwordsFr from 'stopwords-fr';
+import stopwordsEn from 'stopwords-en';
+import { Types } from 'mongoose';
+
+dayjs.extend(utc);
+
+@Injectable()
+export class ReadabilityService {
+  constructor(private db: DbService) {}
+
+  getPages() {
+    return this.db.collections.pagesList
+      .aggregate<{ url: string }>()
+      .limit(5)
+      .project({ _id: 0, url: 1 })
+      .exec();
+  }
+
+  async main() {
+    const pages = await this.getPages();
+
+    for (const { url } of pages) {
+      const fullUrl = `https://${url}`;
+
+      const { data } = await axios.get(fullUrl);
+      const $ = load(data);
+
+      const html = $.html();
+      const main = $('main');
+      const mainContent = main.text();
+
+      const langRegex = /canada\.ca\/(en|fr)/i;
+      const lang = langRegex.exec(fullUrl)[1];
+
+      const preText = mainContent
+        .replace('..', '.')
+        .replace('.', '. ')
+        .slice(0, mainContent.indexOf('defPreFooter'))
+        .replace(/\n/g, '')
+        .replace(/\t/g, '')
+        .replace(/\r/g, '');
+
+      // add periods after bullet points and headings so that flesh kincaid considers them as sentences
+      const html1 = html.replace(/<\/(li|h[1-6]|p)>/g, '.$&');
+
+      const revText = load(html1);
+      revText('table, div.pagedetails, #chat-bottom-bar').remove();
+      const textContent = revText('main').text();
+
+      const postText = textContent
+        .replace(/\.\./g, '. ')
+        .replace(/(\.)\.\s/g, '$1')
+        .replace(/\s+\.\s|\s+\./g, '. ')
+        .replace(/([.]+)(?=$|\s)/g, '. ')
+        .replace(/\n|\t|\r/g, '')
+        .replace(/([.:;!?])\s*\./g, '$1 ')
+        .replace(/\s+/g, ' ');
+
+      const word = readability.lexiconCount(postText);
+      const sentence = this.sentenceCount(postText);
+      let syllables: number;
+      let original_fk: number;
+      let final_fk: number;
+
+      if (lang === 'fr') {
+        syllables = this.syllableCountFr(postText);
+        original_fk = this.fleschKincaidGradeFr(preText);
+        final_fk = this.fleschKincaidGradeFr(postText);
+      } else {
+        syllables = readability.syllableCount(postText);
+        original_fk = readability.fleschKincaidGrade(preText);
+        final_fk = readability.fleschKincaidGrade(postText);
+      }
+
+      // HTML string to be parsed
+      const mainHtml = load(main.html());
+
+      // get all headings and calculate how many words on average between headings
+      const headings = mainHtml('h1, h2, h3, h4, h5, h6');
+      const len_headings = headings.length;
+      const hratio = word / len_headings;
```

Comment:
```suggestion
      const avg_words_per_header = word / len_headings;
```
Just to make it a little more clear what these properties are without having to find the comments here (can do find/replace for these)